src/components/templates/order-table/index.tsx

```diff
@@ -0,0 +1,333 @@
+import { RouteComponentProps, useLocation } from "@reach/router"
+import { isEmpty } from "lodash"
+import { useAdminOrders } from "medusa-react"
+import qs from "query-string"
+import React, { useEffect, useReducer, useState } from "react"
+import { usePagination, useTable } from "react-table"
+import { removeNullish } from "../../../utils/remove-nullish"
+import { formatDateFilter } from "../../../utils/time"
+import Spinner from "../../atoms/spinner"
+import Table, { TablePagination } from "../../molecules/table"
+import OrderFilters from "../order-filter-dropdown"
+import useOrderTableColums from "./use-order-column"
+
+type OrderFilter = {
+  open: boolean
+  filter?: string | string[] | null
+}
+
+type OrderFiltersType = {
+  limit: number
+  offset: number
+  fulfillment?: OrderFilter
+  payment?: OrderFilter
+  date?: OrderFilter
+  status?: OrderFilter
+  pageIndex: number
+  q: string
+}
+
+const defaultQueryProps = {
+  expand: "shipping_address",
+  fields:
+    "id,status,display_id,created_at,email,fulfillment_status,payment_status,total,currency_code",
+}
+
+const initialState: OrderFiltersType = {
+  limit: 14,
+  offset: 0,
+  status: { open: false, filter: null },
+  fulfillment: { open: false, filter: null },
+  payment: { open: false, filter: null },
+  date: { open: false, filter: null },
+  pageIndex: 0,
+  q: "",
+}
+
+function orderFiltersReducer(state, action) {
+  switch (action.type) {
+    case "filters":
+      return { ...state, ...action.payload }
+    case "query":
+      return { ...state, q: action.payload, limit: 14, offset: 0 }
+    case "limit":
+      return { ...state, limit: action.payload }
+    case "offset":
+      return { ...state, offset: action.payload }
+    case "pageIndex":
+      return { ...state, pageIndex: action.payload }
+    case "reset":
+      return {
+        ...initialState,
+      }
+    default:
+      return state
+  }
+}
+
+const OrderTable: React.FC<RouteComponentProps> = () => {
+  const location = useLocation()
+
+  const filtersOnLoad = qs.parse(location.search, {
+    arrayFormat: "bracket",
+  })
+
+  const parseUrlFilter = (urlVal) => {
+    if (!urlVal) {
+      return { open: false, filter: null }
+    }
+
+    if (Array.isArray(urlVal)) {
+      return { open: true, filter: urlVal[0] }
+    } else {
+      return { open: true, filter: urlVal }
+    }
+  }
+
+  const offs = parseInt(filtersOnLoad?.offset) || 0
+  const lim = parseInt(filtersOnLoad.limit) || 14
+  initialState.pageIndex = offs / lim
+
+  const [state, dispatch] = useReducer(orderFiltersReducer, initialState)
+  const [tempState, setTempState] = useState(initialState)
```

Comment:
Temp state is used to hold filters until "Apply" is clicked. This is needed, since changes to the reducer state will trigger rerenders.